docs/pages/project/contributing/contributing-rego.md

```diff
@@ -0,0 +1,42 @@
+category: contributing
+list: include
+---
+
+## Background
+Meshery has a built-in policy engine, based on [Open Policy Agent (OPA)](https://www.openpolicyagent.org/docs/latest/). Meshery uses the [Rego](https://www.openpolicyagent.org/docs/latest/policy-language/) query language to create these [policies](https://docs.meshery.io/concepts/logical/policies). 
+
+## Prerequisites
+To start contributing to Meshery Policy Engine, make sure you have [OPA CLI](https://www.openpolicyagent.org/docs/latest/#running-opa) installed on your system. You will also need to clone the [Meshery Server](https://github.com/meshery/meshery/) project from Github.
+
+## Components of Meshery Policy Engine
+
+Rego policies are the declarative logic behind the policy engine. These policies define how the underlying engine understands relationships within Meshery Designs. Rego policies can be found [here](https://github.com/meshery/meshery/tree/master/server/meshmodel/kubernetes/v1.25.2/v1.0.0/policies) in the Meshery Github repository.
+
+Currently the Rego policies are invoked from the Go code in Meshery Server. This requires configuring the OPA context as seen [here](https://github.com/meshery/meshkit/blob/master/models/meshmodel/core/policies/rego_policy_relationship.go).
+
+The Meshery Policy Engine is invoked by calling the evaluation endpoint defined [here](https://github.com/meshery/meshery/blob/master/server/handlers/policy_relationship_handler.go). This endpoint is passed a Meshery Design in JSON which it passes to the Meshery Policy Engine for validation.
+
+## Working with Meshery Policy Engine
+
+Working with the Meshery Policy Engine can feel complex because of the large number of different relationships a Meshery Design might contain. Contributors must then be careful to test any changes thoroughly to avoid unexpected results. When working with the Rego policies themselves it makes testing and development easier to be able to test the policies directly without running the entire Meshery Server. When working on the policies keep the following in mind:
+
+- Since we will be running these policies using the OPA CLI directly without the Go bootstrap code we need to provide the necessary context for the policy engine. This means providing a data structure containing the valid Meshery Relationships the policy engine will validate against. **Currently there is no automated process for getting this relationship data.**
+- The input to the policy engine is a Meshery Design as a JSON data structure. When testing the Meshery Policy Engine you will need to provide this design as input. If you are working on a bug you will need to get the design from the issue or reporter.
+
+## Executing OPA 
+
+Now that you have the necessary prerequisites you can try running the Rego policies locally. From your local version of Meshery Server navigate to the policies directory:
+{% capture code_content %}cd server/meshmodel/kubernetes/v1.25.2/v1.0.0/policies{% endcapture %}
```

Comment:
@ctcarrier the path shouldn't be hard coded in my opinion since we would be moving them inside meshery-core directory.

Reply:
Totally agree! One solution here would be to add a task to the Makefile to execute policies directly. Or is there perhaps a better place to script something like this?